middle_end/flambda2/kinds/flambda_kind.ml

```diff
@@ -839,6 +839,8 @@ module With_subkind = struct
               match Tag.Scannable.create tag with
               | Some tag ->
                 let shape_and_fields : Block_shape.t * _ =
+                  (* CR mshinwell/vlaviron: In both of these cases it would be
+                     nice to propagate immediacy information. *)
```

Comment:
Actually, I believe that `from_lambda_value_kind` takes care of that, the only remaining case where we lose immediacy is the flat suffix of mixed blocks.
(There are other places where we lose immediacy by computing a `kind` instead of a `kind_with_subkind`, but at least that is explicit.)

Reply:
See discussion elsewhere